packages/remix-server-runtime/server.ts

```diff
@@ -307,14 +317,18 @@ function findParentBoundary(
   routeId: string,
   error: any
 ): string {
-  let route = routes[routeId];
+  // Fall back to the root route if we don't match any routes, since Remix
+  // has default error/catch boundary handling.  This handles the case where
+  // react-router doesn't have a matching "root" route to assign the error to
+  // so it returns context.errors = { __shim-404-route__: ErrorResponse }
+  let route = routes[routeId] || routes["root"];
   let isCatch = isRouteErrorResponse(error);
   if (
-    (isCatch && route.module?.CatchBoundary) ||
-    (!isCatch && route.module?.ErrorBoundary) ||
+    (isCatch && route.module.CatchBoundary) ||
+    (!isCatch && route.module.ErrorBoundary) ||
```

Comment:
This was due to the `__shim-404-route_` stuff.  We now find the root route, but also made it defensive to fall back on it if we don't find one